js/filters/localeOrderBy_filter.js

```diff
@@ -10,30 +10,45 @@ angular.module('contactsApp')
 		});
 
 		arrayCopy.sort(function (a, b) {
-			var valueA = a[sortPredicate];
-			if (angular.isFunction(valueA)) {
-				valueA = a[sortPredicate]();
-			}
-			var valueB = b[sortPredicate];
-			if (angular.isFunction(valueB)) {
-				valueB = b[sortPredicate]();
-			}
 
-			if (angular.isString(valueA)) {
-				return !reverseOrder ? valueA.localeCompare(valueB) : valueB.localeCompare(valueA);
-			}
+			// Did we pass multiple sorting options? If not, create an array abyway.
+			sortPredicate = angular.isArray(sortPredicate) ? sortPredicate: [sortPredicate];
+			// Let's test the first sort and continue if no sort occured
+			for(var i=0; i<sortPredicate.length; i++) {
+				var sortBy = sortPredicate[i];
 
-			if (angular.isNumber(valueA) || typeof valueA === 'boolean') {
-				return !reverseOrder ? valueA - valueB : valueB - valueA;
-			}
+				var valueA = a[sortBy];
+				if (angular.isFunction(valueA)) {
+					valueA = a[sortBy]();
+				}
+				var valueB = b[sortBy];
+				if (angular.isFunction(valueB)) {
+					valueB = b[sortBy]();
+				}
 
-			if (angular.isArray(valueA)) {
-				if (valueA[0] === valueB[0]) {
-					return !reverseOrder ? valueA[1].localeCompare(valueB[1]) : valueB[1].localeCompare(valueA[1]);
+				// Start sorting
+				if (angular.isString(valueA)) {
+					if(valueA !== valueB) {
+						return !reverseOrder ? valueA.localeCompare(valueB) : valueB.localeCompare(valueA);
```

Comment:
I suggest removing the `!` and flipping the both expressions. (also lines 38, 45 and 48)